packages/flutter_tools/lib/src/ios/code_signing.dart

```diff
@@ -116,229 +159,787 @@ Future<Map<String, String>?> getCodeSigningIdentityDevelopmentTeamBuildSetting({
+    _config.setValue(kConfigCodeSignCertificate, identity);
   }
 
-  final List<String> validCodeSigningIdentities =
-      findIdentityStdout
-          .split('\n')
-          .map<String?>((String outputLine) {
-            return _securityFindIdentityDeveloperIdentityExtractionPattern
-                .firstMatch(outputLine)
-                ?.group(1);
-          })
-          .where(_isNotEmpty)
-          .whereType<String>()
-          .toSet() // Unique.
-          .toList();
-
-  final String? signingIdentity = await _chooseSigningIdentity(
-    validCodeSigningIdentities,
-    logger,
-    config,
-    terminal,
-    shouldExitOnNoCerts,
-  );
+  void _saveProvisioningProfile(_ProvisioningProfile profile) {
+    _logger.printStatus('Provisioning Profile "${profile.name}" saved.');
+    _config.setValue(kConfigCodeSignProvisioningProfile, profile.filePath);
+  }
 
-  // If none are chosen, return null.
-  if (signingIdentity == null) {
-    return null;
+  /// Validates that command-line tools `security` and `openssl` are available.
+  Future<bool> _validateCodeSignSearchTools({bool printError = false}) async {
+    // If the user's environment is missing the tools needed to find and read
+    // certificates, abandon. Tools should be pre-equipped on macOS.
+    if (!await _processUtils.exitsHappy(const <String>['which', 'security']) ||
+        !await _processUtils.exitsHappy(const <String>['which', 'openssl'])) {
+      if (printError) {
+        _logger.printError('Unable to validate code-signing tools.');
+      } else {
+        _logger.printTrace('Unable to validate code-signing tools.');
+      }
+      return false;
+    }
+    return true;
   }
 
-  logger.printStatus('Developer identity "$signingIdentity" selected for iOS code signing');
+  /// Get list of code-signing identities.
+  Future<List<String>> _getSigningIdentities() async {
+    String findIdentityStdout;
+    try {
+      findIdentityStdout =
+          (await _processUtils.run(<String>[
+            'security',
+            'find-identity',
+            '-p',
+            'codesigning',
+            '-v',
+          ], throwOnError: true)).stdout.trim();
+    } on ProcessException catch (error) {
+      _logger.printError('Unexpected failure from find-identity: $error.');
+      return <String>[];
+    }
 
-  final String? signingCertificateId = _securityFindIdentityCertificateCnExtractionPattern
-      .firstMatch(signingIdentity)
-      ?.group(1);
+    final List<String> validCodeSigningIdentities =
+        findIdentityStdout
+            .split('\n')
+            .map<String?>((String outputLine) {
+              return _securityFindIdentityDeveloperIdentityExtractionPattern
+                  .firstMatch(outputLine)
+                  ?.group(1);
+            })
+            .where(_isNotEmpty)
+            .whereType<String>()
+            .toSet() // Unique.
+            .toList();
+    return validCodeSigningIdentities;
```

Comment:
Nit: consider just returning directly here. I don't feel strongly about this, feel free to skip.

Something like:

```dart
return findIdentityStdout
  .split('\n')
  ...
```